### PR TITLE
Minor memory improvements

### PR DESCRIPTION
**Pull request by ShFil119**
Bigger tasks still to do:
- AIGraph
- objects of GameWorld
- GameObject (itself)
- Animation

### CODE REVIEW

File: rwengine/src/script/SCMFile.hpp [hunk 1]
```diff
@@ -18,19 +18,48 @@ class SCMFile {
 
     SCMFile() = default;
 
-    ~SCMFile() {
-        delete[] _data;
+    SCMFile(SCMFile&& info)
```
[the remark this answers is not in the document]
Comment (ShFil119) [reply rwengine/src/script/SCMFile.hpp]: Nope, written without rethinking.

File: rwengine/src/render/DebugDraw.hpp [hunk 2]
```diff
@@ -43,8 +43,8 @@ class DebugDraw final : public btIDebugDraw {
 
     std::vector<GeometryVertex> lines;
     size_t maxlines;
-    GeometryBuffer *lineBuff;
-    DrawBuffer *dbuff;
+    std::unique_ptr<GeometryBuffer> lineBuff = std::make_unique<GeometryBuffer>();
```
Comment (ShFil119) [inline rwengine/src/render/DebugDraw.hpp]: just wonder do we need dynamic allocation here?

Comment (danhedron) [reply rwengine/src/render/DebugDraw.hpp]: Doesn't seem like it, it only lives as long as the class anyway.

Comment (ShFil119) [reply rwengine/src/render/DebugDraw.hpp]: So it's "to do" later. ;)

File: rwcore/data/Clump.cpp [hunk 3]
```diff
@@ -113,10 +113,10 @@ void Clump::recalculateMetrics() {
     }
 }
 
-Clump* Clump::clone() const {
+ClumpPtr Clump::clone() const {
```
[the remark this answers is not in the document]
Comment (ShFil119) [reply rwcore/data/Clump.cpp]: You mean `setClump(ClumpPtr(getModel()->clone()));` to `setClump(getModel()->clone());`

[the remark this answers is not in the document]
Comment (danhedron) [reply rwcore/data/Clump.cpp]: Yes.

[the remark this answers is not in the document]
Comment (ShFil119) [reply rwcore/data/Clump.cpp]: Fixed.

**Pull request merged**